documentation/docs/specs/revocation_bitmap_2022.md

```diff
@@ -0,0 +1,196 @@
+7. Return `true` if **revoked** is 1, `false` otherwise.
+
+## Test Vectors
+
+This section provides test vectors to validate implementations against.
+
+### Test Vector 1
+
+The following data URL decodes to a bitmap of length 0 where no index is revoked:
+
+`"data:application/octet-stream;base64,ZUp5ek1tQUFBd0FES0FCcg=="`
+
+### Test Vector 2
+
+The following data URL decodes to a bitmap of length 3 where indices `5`, `398`, and `67000` are revoked:
+
+`"data:application/octet-stream;base64,ZUp5ek1tQmdZR0lBQVVZZ1pHQ1FBR0laSUdabDZHUGN3UW9BRXVvQjlB"`.
+
+### Test Vector 3
+
+The following data URL decodes to a bitmap of length 16384 where all indices are revoked:
+
+`"data:application/octet-stream;base64,ZUp6dHhERVJBQ0FNQkxESEFWS1lXZkN2Q3E0MmFESmtyMlNrM0ROckFLQ2RBQUFBQUFBQTMzbGhHZm9q"`
+
+## Rationale
+
+This section describes the rationale behind some of the design decisions of this specification.
+
+### Compression
+
+Considering that messages published to the Tangle cannot exceed [32 KiB](https://github.com/iotaledger/tips/blob/main/tips/TIP-0006/tip-0006.md#message-validation) in size, and that larger messages have increased requirements, the use of compression was assessed.
+
+ZLIB [[RFC 1950](https://datatracker.ietf.org/doc/html/rfc1950)] was chosen for having a free and open source software licence and being one of the most widely used compression schemes, which enhances the accessibility of this specification. Some other assessed algorithms had only marginally better compression ratios but far fewer existing implementations in different programming languages.
+
+### Compressed Bitstring vs. Roaring Bitmap
+
+Because of their space efficiency, a roaring bitmap is preferred for representing a bitmap in-memory. To avoid the dependency on roaring bitmap, we considered to use a compressed bitstring as the serialization format. However, serialization of such a bitstring is 2-3x slower compared to roaring's serialization format, which becomes an issue on resource-constrained devices (e.g. smartphones) or in web browsers.
+
+### Comparison to `RevocationList2020`
+
+The [RevocationList2020 specification](https://w3c-ccg.github.io/vc-status-rl-2020/) describes a revocation mechanism using a verifiable credential that contains a bitmap, similar to the `RevocationBitmap2022` approach. The credential is hosted outside of the DID document and the verifier thus needs to fetch it from an external resource, likely one controlled by the issuer. This has privacy implications as the issuer can track where a fetch request for the credential came from and infering who the credential was verified by and for what purpose. The issuer can also potentially infer which credential was checked. Because `RevocationBitmap2022` is embedded in the DID document, which can be obtained without the issuer's knowledge, this approach does not suffer from those shortcomings. See also the [privacy considerations](#privacy-considerations).
+
```

Comment:
Maybe also mention the minimum size of `RevocationList2020`.

```suggestion

Another difference is that [RevocationList2020](https://w3c-ccg.github.io/vc-status-rl-2020/) specifies a minimum initial size of 131,072 for its bitstring, to mitigate the potential for correlating individuals when few credentials have been issued. `RevocationBitmap2022` uses 
a roaring bitmap instead of a bitstring, meaning there is no minimum size and the maximum index is not fixed (apart from the upper bound of an unsigned 32-bit integer). This means the bitmap cannot be used to correlate small populations without more information not present in the bitmap itself. However, both schemes still reveal publicly how many credentials have been revoked, which could be used to infer other information if more knowledge about how an issuer assigns credential revocation indexes is known.

```